elifetools/utils.py

```diff
@@ -312,20 +342,29 @@ def extract_nodes(soup, nodename, attr=None, value=None):
         return [tag for tag in soup.descendants if tag.name in nodename]
 
 
-def extract_first_node(soup, nodename, attr=None, value=None):
-    "return the first node of name nodename and optionally matching an attribute value"
+def lazy_extract_nodes(soup, nodename, attr=None, value=None):
+    """
+    Returns a list of tags (nodes) from the given soup matching the given nodename.
+    If an optional attribute and value are given, these are used to filter the results
+    further."""
     # convert string value to list
+    if isinstance(nodename, str):
+        nodename = [nodename]
     if attr is not None and value is not None:
         # filter nodes by tag namd and attribute name
         for tag in soup.descendants:
-            if tag.name == nodename and tag.get(attr) == value:
-                return tag
+            if tag.name in nodename and tag.get(attr) == value:
```

Comment:
That is a little different. I remember introducing `extract_first_node()` recently, and all the calls to it only supplied a single tag name. Either equal match, or in a list of one value, it will be returning the same result.